lib/northbound_grpc.cpp

```diff
@@ -867,19 +879,24 @@ static void *grpc_pthread_start(void *arg)
 
 static int frr_grpc_init(unsigned long *port)
 {
+	fpt = frr_pthread_new(&attr, "frr-grpc", "frr-grpc");
+	fpt->data = (void *)port;
+
 	/* Create a pthread for gRPC since it runs its own event loop. */
-	if (pthread_create(&grpc_pthread, NULL, grpc_pthread_start, port)) {
+	if (frr_pthread_run(fpt, NULL) < 0) {
 		flog_err(EC_LIB_SYSTEM_CALL, "%s: error creating pthread: %s",
 			 __func__, safe_strerror(errno));
 		return -1;
 	}
-	pthread_detach(grpc_pthread);
+	pthread_detach(fpt->thread);
```

Comment:
We shouldn't need `pthread_detach()` anymore since we're calling `frr_pthread_destroy()` on `frr_grpc_finish()`.

Reply:
`frr_pthread_destroy` just frees the allocated structure, it doesn't do anything with the internal pthread resources, which still technically need to be freed. But currently we don't clean those up anyway; the only time we kill this thread is when the program exits, so I don't think it makes much difference whether it's called or not.